src/geojson/geojson_intf.ml

```diff
@@ -66,7 +66,7 @@ module type Json_conv = sig
   type json
 
   val of_json : json -> (t, [ `Msg of string ]) result
-  val to_json : t -> json
+  val to_json : ?bbox:float array option -> t -> json
```

Comment:
This is not quite right, you have doubly wrapped the `bbox` function as an `option option` by specifying `?` and that it is an `option`... if that makes sense.

This has then proliferated to all of the other functions where you have:

```ocaml
let f ?bbox t =  
  match bbox with 
    | Some x -> ...
```

You shouldn't need the pattern-matching.

Reply:
> This is not quite right, you have doubly wrapped the `bbox` function as an `option option` by specifying `?` and that it is an `option`... if that makes sense.
> 
> This has then proliferated to all of the other functions where you have:
> 
> ```ocaml
> let f ?bbox t =  
>   match bbox with 
>     | Some x -> ...
> ```
> 
> You shouldn't need the pattern-matching.

Hi, good morning Sir @patricoferris.
I don't know if I'm missing something but I feel it is actually right . If it were not, the build would have failed, indicating an error that there is a mismatch between the geojson.ml file and it's interface definition in geojson_intf.ml.

Yes, by specifying `?`, I wrapped the bbox with `option`. Meanwhile, the function `bbox_to_json_or_null` expects bbox to be of `float array option`. Ideally, this means that because of the 'wrapping', bbox will be converted in `to_json` as `float array option option`
The wrapping `option` (outer `option`) is strictly for checking if `bbox` was received or not,

Therefore, if I match bbox with -> `Some x`, it would translate to `Some (float array option)`, meaning that the argument was indeed passed.
Therefore, `x` has type `float array option` which we then pass to `bbox_to_json_or_null` which is expecting the same type.

However, if I pass `bbox` from `to_json` directly to `bbox_to_json_or_null`(which is expecting type `float array option`) without pattern matching, the compiler would assume that the `option` (in `bbox` with type of `float array option`) was induced by `?`.
The implication is that to_json would be thought of to have received an argument of `float array` which was then wrapped by `option` as a result of `?`; as opposed to the ideal `float array option` we are meant to recieve in `to_json`.

Reply:
To be honest I'm not quite following your argument ^^" 

Let's take an example: 

```ocaml
let to_json ?bbox t =
  match bbox with
  | Some x ->
    J.obj
      [
        ("type", J.string "FeatureCollection");
        ("bbox", bbox_to_json_or_null @@ x);
        ("features", J.list to_json t);
      ]
  | None ->
      J.obj
        [
          ("type", J.string "FeatureCollection");
          ("features", J.list to_json t);
        ]
```

So here because `bbox_to_json_or_null` takes an optional `float array` the argument to `to_json` is `float array option option`. What I had in mind really was something like the following:

```ocaml
let bbox_to_json_or_empty bbox =
      Option.(if is_some bbox then [ "bbox", J.array J.float (get bbox) ] else [])

let to_json ?bbox position =
        J.obj
          ([
            ("type", J.string typ);
            ("coordinates", Position.to_json position);
          ] @ (bbox_to_json_or_empty @@ bbox))
```

Reply:
> To be honest I'm not quite following your argument ^^"
> 
> Let's take an example:
> 
> ```ocaml
> let to_json ?bbox t =
>   match bbox with
>   | Some x ->
>     J.obj
>       [
>         ("type", J.string "FeatureCollection");
>         ("bbox", bbox_to_json_or_null @@ x);
>         ("features", J.list to_json t);
>       ]
>   | None ->
>       J.obj
>         [
>           ("type", J.string "FeatureCollection");
>           ("features", J.list to_json t);
>         ]
> ```
> 
> So here because `bbox_to_json_or_null` takes an optional `float array` the argument to `to_json` is `float array option option`. What I had in mind really was something like the following:
> 
> ```ocaml
> let bbox_to_json_or_empty bbox =
>       Option.(if is_some bbox then [ "bbox", J.array J.float (get bbox) ] else [])
> 
> let to_json ?bbox position =
>         J.obj
>           ([
>             ("type", J.string typ);
>             ("coordinates", Position.to_json position);
>           ] @ (bbox_to_json_or_empty @@ bbox))
> ```

Pardon my ignorance Sir.
I think I get what you mean now.  Thank you for making me understand.
I will incorporate this into my work.